lib/index.js

```diff
@@ -205,6 +205,16 @@ function formatItems(track) {
       SKU: product.sku()
     });
 
+    // ensure unique $event_id is associated with each Ordered Product event by combining Order Completed
+    //  order_id and product's productId or SKU
+    if (product.productId()) {
```

Comment:
if you are having issue with coercion of undefined values, I'd do this maybe something like this

```
var identifier = product.productId() || product.id() || product.sku();
item.$event_id = orderId + '_' + identifier;
```

@brennan